src/test/eventSchemas/providers/schemasDataProvider.test.ts

```diff
@@ -0,0 +1,141 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as assert from 'assert'
+import { RegistrySummary, SchemaSummary } from 'aws-sdk/clients/schemas'
+import * as sinon from 'sinon'
+import { Cache, regionRegistryMap, SchemasDataProvider } from '../../../eventSchemas/providers/schemasDataProvider'
+import { MockSchemaClient } from '../../shared/clients/mockClients'
+import { asyncGenerator } from '../../utilities/collectionUtils'
+
+describe('schemasDataProvider', () => {
+    let sandbox: sinon.SinonSandbox
+    let dataProviderObject: SchemasDataProvider
+    let regionDataList: regionRegistryMap[]
+    let cacheData: Cache
+
+    beforeEach(async () => {
+        sandbox = sinon.createSandbox()
+        regionDataList = []
+        cacheData = new Cache(regionDataList)
+        dataProviderObject = new SchemasDataProvider(testCredentials, cacheData)
+        sandbox.stub(schemaClient, 'listRegistries').returns(asyncGenerator([registrySummary1, registrySummary2]))
+        sandbox.stub(schemaClient, 'listSchemas').returns(asyncGenerator([schemaSummary, schemaSummary2]))
+    })
+
+    afterEach(() => {
+        sandbox.restore()
+    })
+
+    const TEST_REGION = 'testRegion'
+    const TEST_REGISTRY = 'testRegistry'
+    const TEST_REGISTRY2 = 'testRegistry2'
+    const TEST_SCHEMA = 'testSchema'
+    const TEST_SCHEMA2 = 'testSchema2'
+    const registrySummary1 = { RegistryName: TEST_REGISTRY }
+    const registrySummary2 = { RegistryName: TEST_REGISTRY2 }
+    const schemaSummary = { SchemaName: TEST_SCHEMA }
+    const schemaSummary2 = { SchemaName: TEST_SCHEMA2 }
+    const schemaClient = new MockSchemaClient(TEST_REGION)
+    const testCredentials = ({} as any) as AWS.Credentials
+
+    describe('getRegistries', () => {
+        it('should return registries for given region', async () => {
+            const registryNames = await dataProviderObject.getRegistries(TEST_REGION, schemaClient)
+            assert.ok(registryNames!.length === 2, 'unexpected number of registries returned')
+            assert.strictEqual(registryNames![0], TEST_REGISTRY, 'TEST_REGISTRY name should match')
+            assert.strictEqual(registryNames![1], TEST_REGISTRY2, 'TEST_REGISTRY2 name should match')
+        })
+
+        it('should retain results once it is queried ', async () => {
```

Comment:
Based on the description, it sounds like you are testing that the cache is being used, but the test doesn't do this. Can you double check this test please?